Lib/tomllib/_parser.py

```diff
@@ -71,7 +71,12 @@ def loads(s: str, /, *, parse_float: ParseFloat = float) -> dict[str, Any]:  # n
 
     # The spec allows converting "\r\n" to "\n", even in string
     # literals. Let's do so to simplify parsing.
-    src = s.replace("\r\n", "\n")
+    try:
+        src = s.replace("\r\n", "\n")
+    except (AttributeError, TypeError):
+        raise TypeError(
+            f"Expected str object, not '{type(__s).__qualname__}'"
```

Comment:
Thanks for the review! Is the use of a dunder attr instead of a builtin function, and repr to get quotes, based on some style guide? It's not obvious to me why it's better.

Reply:
I'd prefer to keep `type()`, as the `__class__` attribute may be shadowed. No opinion on `!r` vs. quotes.

Reply:
Oh yes. Err, it's just out of habit. No worries, you can keep `type(s).__qualname__` (I like using !r because I don't need to wonder about quotes in general).